### DISCUSSION TITLE
git:from-image and deprecate tags plugin

### DISCUSSION THREAD

**Discussion by coaperator** [dokku/dokku#4912] [Q&A] [answered]
hi, I have a pipeline and I deploy the application
`dokku tags:deploy $ APP_NAME $ APP_TAG`
in the new version of dokku tags: deploy has been removed
what needs to be written to work with git:from-image?
I looked at the documentation but didn't quite get it

**Comment by Cellane** [comment, marked as answer]
Hi! The equivalent of the old `dokku tags deploy <app-name> <image-tag>` is `dokku git:from-image <app-name> <fully-qualified-image-name>`. For example, when deploying images hosted from Docker Hub, what was previously a set of commands:

```
docker image pull livebook/livebook:0.3.1
docker image tag livebook/livebook:0.3.1 dokku/livebook:0.3.1
dokku tags deploy livebook 0.3.1
```

… is now just one simple command: `dokku git:from-image livebook livebook/livebook:0.3.1`.

**Reply by coaperator** [reply]
that is, the commands
```
docker image pull livebook/livebook:0.3.1
docker image tag livebook/livebook:0.3.1 dokku/livebook:0.3.1
```
are no longer needed?

I understood correctly that the command
`dokku git:from-image livebook livebook/livebook:0.3.1`
download an image, assign a tag, and create an app if it doesn't exist?

**Reply by Cellane** [reply]
> that is, the commands […] are no longer needed?

Exactly right!

> I understood correctly that the command `dokku git:from-image livebook livebook/livebook:0.3.1` download an image, assign a tag, and create an app if it doesn't exist?

Almost! It downloads an image, assigns a tag[^1] and deploys it to an already existing app. If I’m not mistaken, it will not create an app for you though (I think!), you still have to run `dokku apps:create <app-name>` once at the beginning.

[^1]: realistically, internally it works a bit differently actually. Dokku downloads the image you requested, builds a new image based on the downloaded image (it doesn’t just retag the existing image, it literally builds a new one with a generated `Dockerfile` that would begin with the line `FROM livebook/livebook:0.3.1` in the above example), assigns some labels (incl. a label that points back at the original image, if needed), and then tags the newly build image under the `dokku/` namespace.

**Reply by coaperator** [reply]
thank you very much!
